plugins/by-name/chadtree/default.nix

```diff
@@ -502,11 +498,21 @@ in
       };
     in
     mkIf cfg.enable {
-      extraPlugins =
-        [ cfg.package ]
-        ++ (optional (
-          cfg.iconsPackage != null && (cfg.theme == null || cfg.theme.iconGlyphSet == "devicons")
-        ) cfg.iconsPackage);
+      # TODO: added 2024-09-20 remove after 24.11
+      plugins.web-devicons.enable = mkOverride 1490 true;
+      warnings =
+        optional
+          (
+            (cfg.theme == null || cfg.theme.iconGlyphSet == "devicons")
+
+            && options.plugins.web-devicons.enable.highestPrio == 1490
+          )
+          ''
+            Nixvim (plugins.chadtree) `web-devicons` automatic installation is deprecated.
+            Please use `plugins.web-devicons.enable`.
+          '';
```

Comment:
Should this warning give any hints for how to use `mini.icons` instead of `web-devicons`? Or if we had an assertion, perhaps those kinda notes could go there?

Not that I'm implying I do or don't want an assertion...

Reply:
If I do the mkRemovedOptionModule then it will contain those instructions.

Reply:
> Please use `plugins.web-devicons` or `plugins.mini.modules.icons` with `plugins.mini.mockDevIcons` instead.